torch/distributed/nn/api/remote_module.py

```diff
@@ -143,6 +158,114 @@ def __init__(
             method = torch.jit.export(method)
             setattr(self, method_name, types.MethodType(method, self))
 
+    def _raise_not_supported(self):
+        raise ValueError("Method not supported for RemoteModule")
+
+    def register_buffer(
+        self, name: str, tensor: Optional[Tensor], persistent: bool = True
+    ) -> None:
+        _raise_not_supported()
+
+    def register_parameter(self, name: str, param: Optional[Parameter]) -> None:
+        _raise_not_supported()
+
+    def add_module(self, name: str, module: Optional["Module"]) -> None:
+        _raise_not_supported()
+
+    def apply(self: T, fn: Callable[["Module"], None]) -> T:
+        _raise_not_supported()
+
+    def cuda(self: T, device: Optional[Union[int, device]] = None) -> T:
+        _raise_not_supported()
+
+    def cpu(self: T) -> T:
+        _raise_not_supported()
+
+    def type(self: T, dst_type: Union[dtype, str]) -> T:
+        _raise_not_supported()
+
+    def float(self: T) -> T:
+        _raise_not_supported()
+
+    def double(self: T) -> T:
+        _raise_not_supported()
+
+    def half(self: T) -> T:
+        _raise_not_supported()
+
+    def bfloat16(self: T) -> T:
+        _raise_not_supported()
+
+    def to(self, *args, **kwargs):
+        _raise_not_supported()
+
+    def register_backward_hook(
+        self, hook: Callable[["Module", _grad_t, _grad_t], Union[None, Tensor]]
+    ) -> RemovableHandle:
+        _raise_not_supported()
+
+    def register_forward_pre_hook(self, hook: Callable[..., None]) -> RemovableHandle:
+        _raise_not_supported()
+
+    def register_forward_hook(self, hook: Callable[..., None]) -> RemovableHandle:
+        _raise_not_supported()
+
+    def state_dict(self, destination=None, prefix="", keep_vars=False):
+        _raise_not_supported()
+
+    def load_state_dict(
+        self,
+        state_dict: Union[Dict[str, Tensor], Dict[str, Tensor]],
+        strict: bool = True,
+    ):
+        _raise_not_supported()
+
+    def parameters(self, recurse: bool = True) -> Iterator[Parameter]:
+        _raise_not_supported()
```

Comment:
Maybe this one should mention that we should use `remote_parameters()` instead in the error message.